lib/errors/index.ts

```diff
@@ -0,0 +1,94 @@
+import * as http from 'http';
+import * as rawErrors from './arsenalErrors';
+import * as types from './types';
+
+export * from './types';
+
+/** Mapping used to determine an error type. */
+export type Is = { [Name in types.Name]: boolean };
+/** Mapping of all possible Errors */
+export type Errors = { [Property in keyof types.Names]: ArsenalError };
+
+const entries = Object.entries(rawErrors);
```

Comment:
Entries is only used one time in the generate function below, I would just use it inside that function. Otherwise we keep this value around indefinitely?